### PR TITLE
remove HAPI FHIR references and generate ELM on the fly during fetch …

### PR DESCRIPTION
**Pull request by nmorasb**
…of versioned library

## MADiE PR

Jira Ticket: [MAT-5405](https://jira.cms.gov/browse/MAT-5405)
(Optional) Related Tickets:
[MAT-5593](https://jira.cms.gov/browse/MAT-5593)

### Summary
Remove interface with HAPI FHIR
Update the `/versioned` fetch API for CQL Library to include ELM generated at fetch-timed

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included and Code coverage is verified.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java [hunk 1]
```diff
@@ -135,7 +137,7 @@ public String getLibraryCql(
       @RequestParam String name,
       @RequestParam String version,
       @RequestParam Optional<String> model) {
-    return cqlLibraryService.getVersionedCqlLibrary(name, version, model).getCql();
+    return cqlLibraryService.getVersionedCqlLibrary(name, version, model, null).getCql();
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:140]: does this not need to be protected?

File: src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java [hunk 2]
```diff
@@ -67,9 +53,9 @@ public CqlLibrary createVersion(String id, boolean isMajor, String username, Str
       cqlLibrary.setElmXml(elmJson.getXml());
       // TODO: determine if integration with a different external service is needed, like future
       // madie-dqm-service?
-      if (ModelType.QI_CORE.getValue().equals(cqlLibrary.getModel())) {
-        persistHapiFhirCqlLibrary(cqlLibrary, accessToken);
-      }
+      //      if (ModelType.QI_CORE.getValue().equals(cqlLibrary.getModel())) {
+      //        persistHapiFhirCqlLibrary(cqlLibrary, accessToken);
+      //      }
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java]: maybe we can remove this as well since we removed the method implementation.

**Review by adongare — DISMISSED** [review]

File: src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java (hunk 2 above)
Comment (nmorasb) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java]: removed!

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java (hunk 1 above)
Comment (nmorasb) [reply src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:140]: so, the route would be protected by Okta per the SecurityConfig. However if there is no access token passed down into the getVersionedCqlLibrary method then it will skip generating the ELM as the token is needed for getting the ELM (to make the call to the translator). This is because for just fetching CQL we don't need the ELM, so saving the time

Comment (adongare) [reply src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:140]: oh yeah, that is right. thank you.

**Review by adongare — APPROVED** [review]
